Assets/SEE/XR/Rotator.cs

```diff
@@ -4,185 +4,210 @@
-            GetRotationDistance(rotationAngle);
+            grabInteractor.selectEntered.AddListener(GrabbedBy);
+            grabInteractor.selectExited.AddListener(GrabEnd);
         }
-    }
-    /// <summary>
-    /// Gets the current roation-angle from the controller.
-    /// </summary>
-    /// <returns>The current rotation-angle from the controller.</returns>
-    public float GetInteractorRotation() => interactor.GetComponent<Transform>().eulerAngles.z;
-    /// <summary>
-    /// Determins in which direction and how much the dial/node should be rotated.
-    /// </summary>
-    /// <param name="currentAngle">The current angle from the controller.</param>
-    private void GetRotationDistance(float currentAngle)
-    {
-        if (!requiresStartAngle)
+
+        private void OnDisable()
         {
-            var angleDifference = Mathf.Abs(startAngle - currentAngle);
-            if (angleDifference > angleTolerance)
+            grabInteractor.selectEntered.RemoveListener(GrabbedBy);
+            grabInteractor.selectExited.RemoveListener(GrabEnd);
+        }
+
+        /// <summary>
+        /// This method gets called when the user stops using the dial.
+        /// </summary>
+        /// <param name="args">Event data associated with the event when an Interactor stops selecting an Interactable.</param>
+        private void GrabEnd(SelectExitEventArgs args)
+        {
+            ShouldGetHandRotation = false;
+            firstRotation = true;
+        }
+
+        /// <summary>
+        /// This method gets called when the user begins using the dial.
+        /// </summary>
+        /// <param name="args">Event data associated with the event when an Interactor first initiates selecting an Interactable.</param>
+        private void GrabbedBy(SelectEnterEventArgs args)
+        {
+            interactor = (XRBaseInteractor)GetComponent<XRGrabInteractable>().interactorsSelecting[0];
+            ShouldGetHandRotation = true;
+            startAngle = 0f;
+        }
+
+        private void Update()
+        {
+            if (GlobalActionHistory.Current() != ActionStateTypes.Rotate)
+            {
+                Destroyer.Destroy(gameObject);
+            }
+            if (ShouldGetHandRotation)
+            {
+                float rotationAngle = GetInteractorRotation();
+                GetRotationDistance(rotationAngle);
+            }
+        }
+
+        /// <summary>
+        /// Returns the current rotation-angle from the controller.
+        /// </summary>
+        /// <returns>The current rotation-angle from the controller.</returns>
+        public float GetInteractorRotation() => interactor.transform.eulerAngles.z;
+
+        /// <summary>
+        /// Determines in which direction and how much the dial/node should be rotated.
+        /// </summary>
+        /// <param name="currentAngle">The current angle from the controller.</param>
+        private void GetRotationDistance(float currentAngle)
```

Comment:
Since the method has a return value of `void`, its name should not start with `Get`. Maybe rather something like `DetermineRotationDistance`?